gotpm/cmd/seal_test.go

```diff
@@ -85,16 +97,29 @@ func TestUnsealFail(t *testing.T) {
 	defer tpm2tools.CheckedClose(t, rwc)
 	ExternalTPM = rwc
 
+	// func to be executed between Seal and Unseal in the following tests
+	type intermediateFunc func(io.ReadWriter, []string) error
+	var extendPCR intermediateFunc = func(rw io.ReadWriter, pcrs []string) error {
+		pcr, err := strconv.ParseInt(pcrs[0], 10, 32)
+		if err != nil {
+			return err
+		}
+		pcrToExtend := tpmutil.Handle(pcr)
+		extension := bytes.Repeat([]byte{0xAA}, sha256.Size)
+		return tpm2.PCRExtend(rw, pcrToExtend, tpm2.AlgSHA256, extension, "")
+	}
+
 	tests := []struct {
-		name    string
-		tpmFunc func(io.ReadWriter) error
+		name              string
+		sealPCRs          string
+		certifyPCRs       string
+		tpmFuncs          []intermediateFunc
```

Comment:
If we're always have the exact same code for `tpmFuncs` just move the `intermediteFunc` code into the test for loop, and remove this indirection.

Reply:
I can do that. But in the future we probably need more "intermediate" functions other than extendPCR. For now I can move extendPCR into the loop for simplicity.